arch/simddetect.cpp

```diff
@@ -26,7 +26,7 @@
 #endif // x86 target
 
 #if defined(X86_BUILD)
-# if defined(__linux__) || defined(__MINGW32__) || defined(__APPLE__)
+# if defined(__GNUC__) || defined(__clang__) || defined(__MINGW32__) || defined(__MINGW64__) 
```

Comment:
clang also provides __get_cpuid(). Not sure about ICC.

I don't think it's a bad thing for the list to be inclusive of various compilers expected to work even if some defines are redundant. What do you think?